dpdbaltics.php

```diff
@@ -549,8 +550,8 @@ public function getOrderShippingCostExternal($cart)
             );
         /** @var PriceRuleService $priceRuleService */
         $priceRuleService = $this->getModuleContainer()->get('invertus.dpdbaltics.service.price_rule_service');
-
-        return $priceRuleService->applyPriceRuleForCarrier($cart, $priceRulesIds, $this->context->shop->id);
+        $shippingPriceCalculationService = new ShippingPriceCalculationService();
+        return  $shippingPriceCalculationService->calculate($cart, $priceRuleService, $priceRulesIds);
```

Comment:
You have the cart, pass delivery address and get priceRulesIds inside the service. It's only used in there, less code inside module files -> better testability.